common/src/network.rs

```diff
@@ -3,12 +3,37 @@
 use bitcoin::blockdata::block::{Block, BlockHeader};
 use bitcoin::consensus::params::Params;
 use bitcoin::hash_types::BlockHash;
+use bitcoin::network::constants::ServiceFlags;
 use bitcoin_hashes::hex::FromHex;
 
 use bitcoin_hashes::sha256d;
 
 use crate::block::Height;
 
+/// Peer services supported by nakamoto.
+#[derive(Debug, Copy, Clone)]
+pub enum Service {
+    /// Peers with compact filter support
+    Filters,
+    /// Peers with only block support.
+    Blocks,
+}
```

Comment:
I'm thinking about this more, as I don't think it reads so well currently :thinking: 

Perhaps this would work better:

```rust
enum Services {
    /// Peers with blockchain and compact filter support.
    All,
    /// Peers with only blockchain support.
    Chain,
}
```

Reply:
Changed!

Reply:
Should be `Services` (plural), as the options may involve more than one service.